addons/account/models/account_reconcile_model.py

```diff
@@ -812,19 +812,22 @@ def _get_invoice_matching_amls_candidates(self, st_line, partner):
                 }
 
         # Search without any matching based on textual information.
-        if partner:
+        if self.matching_order == 'new_first':
+            order = 'date_maturity DESC, date DESC, id DESC'
+        else:
+            order = 'date_maturity ASC, date ASC, id ASC'
 
-            if self.matching_order == 'new_first':
-                order = 'date_maturity DESC, date DESC, id DESC'
-            else:
-                order = 'date_maturity ASC, date ASC, id ASC'
+        if not partner:
+            aml_domain += [
+                ('balance', '=', -st_line.amount_residual),
```

Comment:
Wrong currency! You are comparing appels with oranges.
- add a check on the same currency
- match on balance or amount_currency depending the currency of the company/st line

Note: the currency of the statement line is given by: `st_line.foreign_currency_id or st_line.journal_id.currency_id or st_line.company_currency_id`

Reply:
Thanks for the heads up!